examples/common/src/bevygap_shared.rs

```diff
@@ -0,0 +1,89 @@
+/// Registers a channel and replicated resource for server metadata.
+/// This is used to tell the client info about the server they've connected to.
+use bevy::prelude::*;
+#[cfg(feature = "bevygap_server")]
+use bevygap_server_plugin::prelude::*;
+use lightyear::prelude::*;
+
+#[derive(Clone)]
+pub struct BevygapSharedExtensionPlugin;
+
+impl Plugin for BevygapSharedExtensionPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_resource::<ServerMetadata>();
+        app.register_resource::<ServerMetadata>(ChannelDirection::ServerToClient);
+        app.add_channel::<ServerMetadataChannel>(ChannelSettings {
+            mode: ChannelMode::OrderedReliable(ReliableSettings::default()),
+            ..default()
+        });
+
+        #[cfg(feature = "bevygap_server")]
+        app.add_systems(
+            Update,
+            update_server_metadata.run_if(resource_added::<ArbitriumContext>),
+        );
+
+        #[cfg(feature = "bevygap_client")]
+        {
+            app.add_systems(
+                Update,
+                (
+                    on_server_metadata_changed.run_if(resource_changed::<ServerMetadata>),
+                    on_bevygap_state_change
+                        .run_if(state_changed::<bevygap_client_plugin::BevygapClientState>),
+                ),
+            );
+        }
+    }
+}
+
+/// Used to replicate ServerMetadata resource
+#[derive(Channel)]
+pub struct ServerMetadataChannel;
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, Resource)]
+pub struct ServerMetadata {
+    pub location: String,
+    pub fqdn: String,
```

Comment:
Maybe you could add comments to explain what those are

Reply:
will do